### ISSUE TITLE
Control-Shift-<right>

### ISSUE THREAD

**Issue by blackspotent** [Fuco1/smartparens#523]
Hi, there.
I'm new to smartparens. 
I use C-S-<right> or C-S-<left> alot for selecting "words" 
It seems like C-S-<right> is not bound to anything but it doesn't work when I enable smartprents-global-strict-mode.
How can I bring it back? 
Thank you very much for the wonderful utility.
Cheers!
-chris

**Comment by Fuco1** [comment]
Any other package which might interfere?  SP binds those keys to slurping.

**Comment by chrisk414** [comment]
Without smartparens,  "control shift select" works as expected and I don't think there is any conflict. 

I bound to C-<right> to sp-forward-slurp-sexp but I didn't bind C-S-<right> to anything.  

C-S-<right> does something but M-x describe-key C-S-<right> does not reveal it.

Anyway, is there way to bring the original "control shift select" back?
Thanks.

**Comment by Silex** [comment]
@chrisk414: What does `C-h k C-S-<right>` says? is it bound to anything?

**Comment by chrisk414** [comment]
It is used to select words forward.

**Comment by Silex** [comment]
Well, what elisp function is `C-S-right` bound to? `C-h k KEYCOMBO` will tell you.

**Comment by chrisk414** [comment]
This is what it says.  <C-S-right> translates to <C-right> and I can't tell what it really bounds to.
=============================================================
<C-right> (translated from <C-S-right>) runs the command right-word
(found in global-map), which is an interactive compiled Lisp function
in ‘bindings.el’.

It is bound to <C-right>.

(right-word &optional N)

Move point N words to the right (to the left if N is negative).

Depending on the bidirectional context, this may move either forward
or backward in the buffer.  This is in contrast with M-f
and M-b, which see.

Value is normally t.
If an edge of the buffer or a field boundary is reached, point is left there
there and the function returns nil.  Field boundaries are not noticed
if ‘inhibit-field-text-motion’ is non-nil.

**Comment by Fuco1** [comment]
> <C-right> (translated from <C-S-right>)

Well this is wrong.

**Comment by Silex** [comment]
>> <C-right> (translated from <C-S-right>)

> Well this is wrong.

Indeed. Not sure what would trigger it tho, maybe some key modifiers?

**Comment by blackspotent** [comment]
Am I the only one who has this kind of behavior? 
\<C-S-right\> is bound to \<C-right\> and I would like to keep control-shift-select as before.
This is the only show-stopper using smartparens. 
Please help.
Thanks.

**Comment by Silex** [comment]
Heh, actually by me I have the same:

```
<C-right> (translated from <C-S-right>) runs the command right-word
(found in global-map), which is an interactive compiled Lisp function
in ‘bindings.el’.
```

Except by me it works to select words... so the problem is somewhere else now.

Can you please define what you mean with "it doesn't work"? it doesn't move right? it does but doesn't select?

**Comment by blackspotent** [comment]
Hi, Silex, 
That's good to know. 
I'm using Windows gnu emacs 25.0.50.1

Right now, both behaves the same, i.e., sp-forward-slurp-sexp

With smartparens-mode turned off, \<C-S-right\> selects word as expected.

**Comment by chrisk414** [comment]
Any update on this on using Windows? I'm still having the same problem.

**Comment by Fuco1** [comment]
You can try to unbind `C-S-<right>` form `smartparens-mode-map`.

**Comment by blackspotent** [comment]
I tried to unbind the keys as followw.

;; undefine keys
(define-key smartparens-mode-map (kbd "C-S-<right>") nil)
(define-key smartparens-mode-map (kbd "C-S-<left>") nil)

but the behavior is the same.

I think the problem is that it sees C-S-<right> the same as C-<right> in my Windows emacs. 
But without Smartparent mode enabled, C-S-<right> behaves correctly (selects words)
What in Smartparent would affect C-S-<right> behavior?? 
Please help!!!

Thanks.

**Comment by Fuco1** [comment]
Try also setting `sp-base-key-bindings` to nil.  I honestly don't understand what is going on.

**Comment by chrisk414** [comment]
Hi, all!
Little bit of digging, I found some more information.
I'm using key binding from Fuco where he uses C-/<right/>. 
Once C-/<right/> is bound, C-S-/<right/> will then default to C-/<right/>, thus C-S-/<right/> behaves like C-/<right/>. I'm not sure if it happens only on Windows though.
